### PR TITLE
feat: add critical pressure escape for deferred compaction

### PR DESCRIPTION
**Pull request by Tosko4**
## Summary

Implements Stephen's Option C decision from #142: LCM stays cache-friendly, not fully cache-aware, and gets only a narrow critical-pressure escape hatch.

The new `LCM_CRITICAL_BUDGET_PRESSURE_RATIO` setting is disabled by default. When configured, it can bypass existing polite gates for bounded deferred maintenance catch-up and cache-friendly follow-on condensation once prompt pressure reaches the configured ratio.

## Why

Current cache counters are retrospective status data. They do not tell the plugin whether the next prompt mutation will break a hot provider cache, so provider TTL heuristics and provider-family detection do not belong in `hermes-lcm` yet.

## Validation

- [x] Focused validation: deferred maintenance and cache-friendly condensation tests -> `7 passed`
- [x] Focused audit: Codex CLI read-only diff review against Stephen's decision -> no blockers before PR
- [x] Default validation:
  - [x] `pytest tests/test_lcm_core.py tests/test_lcm_engine.py tests/test_packaging_install.py -q` -> `505 passed`
  - [x] `pytest -q` -> `551 passed`
  - [x] `python -m compileall -q .`
  - [x] `python -m py_compile scripts/import_lossless_claw.py`
  - [x] `bash -n scripts/install.sh scripts/update.sh`
  - [x] `git diff --check`

## Notes

This intentionally does not add cache-touch tracking, cache-break tracking, TTL delays, provider/model heuristics, or provider-family detection. The README documents that full cache-aware deferred compaction should wait until Hermes core exposes reliable cache state.

Codex flagged that the critical-pressure bypass also needs to apply after each dynamic leaf pass. The current head mirrors that bypass in the post-pass remainder check and adds regression coverage.

Refs #142

### CODE REVIEW

File: engine.py [hunk 1]
```diff
@@ -654,14 +662,16 @@ def compress(self, messages: List[Dict[str, Any]],
             if self._config.dynamic_leaf_chunk_enabled:
                 working_leaf_chunk_tokens = self._working_leaf_chunk_tokens(raw_tokens_outside_tail)
                 if raw_tokens_outside_tail < working_leaf_chunk_tokens and not force_overflow:
-                    break
+                    if not (deferred_maintenance_active and critical_budget_pressure):
+                        break
```
[the remark this answers is not in the document]
Comment (Tosko4) [reply engine.py:673]: Fixed in 974d202. The critical-pressure bypass now also applies to the post-pass dynamic remainder check, with regression coverage for multi-pass catch-up. Full suite: 551 passed.

**Review by stephenschoettler — COMMENTED** [review]
I did a focused maintainer pass on PR #151 at `974d2022cda4`.

Verdict:
No blockers from this pass. Safe next step: approve after a fresh GitHub state recheck.

Validation:
Ran in clean temporary checkout:
`/tmp/hermes-lcm-pr151.CcrCtu/repo`
HEAD: `974d2022cda4dc2234de80aec8b960a3c94fcc45`
Base: `6774c96e333e529936312ceb3cb22fa5a374ea8c`

Commands/results:
- `python -V` -> Python 3.14.4
- `pytest tests/test_lcm_core.py tests/test_lcm_engine.py tests/test_packaging_install.py -q` -> 505 passed, 177 warnings
- `pytest -q` -> 551 passed, 177 warnings
- `python -m compileall -q .` -> pass
- `python -m py_compile scripts/import_lossless_claw.py` -> pass
- `bash -n scripts/install.sh scripts/update.sh` -> pass
- `git diff --check origin/main...HEAD` -> pass

Warnings were existing-style Python 3.14 `DeprecationWarning: __package__ != __spec__.parent` import warnings; no test failures.

This is a comment review, not an approval.

**Review by stephenschoettler — APPROVED** [review]
I did a focused maintainer pass on PR #151 at `974d2022cda4`.

Verdict:
No blockers from this pass. Safe next step: approve after a fresh GitHub state recheck.

Critical:
None.

Warnings:
None.

Suggestions:
None.

Looks good:
- Matches #142 / PR contract: adds narrow `LCM_CRITICAL_BUDGET_PRESSURE_RATIO`; does not add TTL/cache-touch tracking, provider-family detection, or cache-aware heuristics.
- Default behavior remains disabled at `0.0`.
- Critical pressure bypass is scoped to the intended polite gates:
  - deferred raw-backlog catch-up
  - dynamic post-pass catch-up after Codex’s flagged edge case
  - cache-friendly follow-on condensation suppression
- Catch-up remains bounded by `deferred_maintenance_max_passes`; fresh-tail protection and forced overflow behavior are not loosened.
- Tests cover defaults/env parsing, status surfacing, under-threshold debt drain, dynamic multi-pass catch-up, missing context telemetry, and cache-friendly condensation bypass.
- README documents the cache-friendly-not-cache-aware boundary accurately.

Validation:
Ran in clean temporary checkout:
`/tmp/hermes-lcm-pr151.CcrCtu/repo`
HEAD: `974d2022cda4dc2234de80aec8b960a3c94fcc45`
Base: `6774c96e333e529936312ceb3cb22fa5a374ea8c`

Commands/results:
- `python -V` -> Python 3.14.4
- `pytest tests/test_lcm_core.py tests/test_lcm_engine.py tests/test_packaging_install.py -q` -> 505 passed, 177 warnings
- `pytest -q` -> 551 passed, 177 warnings
- `python -m compileall -q .` -> pass
- `python -m py_compile scripts/import_lossless_claw.py` -> pass
- `bash -n scripts/install.sh scripts/update.sh` -> pass
- `git diff --check origin/main...HEAD` -> pass

Warnings were existing-style Python 3.14 `DeprecationWarning: __package__ != __spec__.parent` import warnings; no test failures.

Next action:
- Approving this head. I will merge after a final state check.
